### ISSUE TITLE
bug: Fetch interceptor conflicts with some website

### ISSUE THREAD

**Issue by NicodeSS** [requestly/requestly#2001]
### Describe your issue?

Some websites have their fetch interceptor to handle specific logic, such as CSRF validation, and this lead to a conflict with the Requestly plugin. For example, one such website is [juejin.cn](https://juejin.cn). When I view this website, the fetch request will fail and throw an error as shown below:
```plaintext
Uncaught (in promise) TypeError: Cannot set properties of undefined (setting '_method')
    at J.XMLHttpRequest.open (ajaxRequestInterceptor.ps.js:1:11513)
    at r.open (sdk-glue.js:1:83550)
    at 3ce5b98.js:3:2260286
    at new Promise (<anonymous>)
    at e.value (3ce5b98.js:3:2258448)
    at e.value (3ce5b98.js:3:2257742)
    at window.fetch (3ce5b98.js:3:2262372)
    at e (bdms.js:1:214165)
    at e (bdms.js:1:213135)
    at window.fetch (captcha.js:1:32463)
```

### Repro steps

1. Install and enable the latest version of Requestly
2. Visit [juejin.cn](https://juejin.cn)
3.  The page can't be loaded correctly and the console will report `Cannot set properties of undefined (setting '_method')` error

### What Requestly tool were you using?

- [X] Extension
- [ ] Desktop-App
- [ ] Android SDK
- [ ] Selenium

### Your Environment

MacOS 14.2.1 Chrome 126

### Requestly Version

v24.7.23

### Error screenshot

<img width="1621" alt="Snipaste_2024-07-25_15-41-34" src="https://github.com/user-attachments/assets/d916d2d7-4bdd-4d1f-88c9-e2a60dd2b766">

**Comment by nafees87n** [comment]
Hi
You can add this website to blocklist so that Requestly would not have any effect on that website.
Please refer the documentation [here](https://docs.requestly.com/general/blocklist)

**Comment by NicodeSS** [comment]
Thanks for the temporary solution.
However, it's certainly a bug that happens unexpectedly from time to time. is it possible to enhance the plugin's compatibility so that this kind of problem can be solved at the source?

**Comment by NicodeSS** [comment]
Thanks for the temporary solution.
However, it's certainly a bug that happens unexpectedly from time to time. is it possible to enhance the plugin's compatibility so that this kind of problem can be solved at the source?

**Comment by wrongsahil** [comment]
https://github.com/requestly/requestly/issues/1980 Seems to be related to this

**Comment by wrongsahil** [comment]
Hi @NicodeSS, We've released a fix in the latest version, v24.7.29. Please follow [these steps](https://docs.requestly.com/guides/how-to-update-requestly-extension-manually) to update the extension manually. Let us know if this resolves the issue.

**Comment by NicodeSS** [comment]
It seems that the fix looks good to me. There are no errors in these websites now.
